src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupDebug.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    /// <summary>
+    /// <see cref="System.Diagnostics.Debug"/> replacement for the startup path.
+    /// It's not safe to use the full-blown Debug class during startup because big chunks
+    /// of the managed execution environment are not initialized yet.
+    /// </summary>
+    static class StartupDebug
+    {
+        [Conditional("DEBUG")]
+        public static void Assert(bool condition)
```

Comment:
`System.Private.CoreLib` has its own private Debug.Assert implementation. It may be better to fix that instead of introducing yet another private Debug.Assert: have a flag that says whether enough of the startup code run. If the flag is not set, just A/V. If the flag is set, use full assert implementation.